moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -187,8 +189,13 @@ private Object cacheKey(Type type, Set<? extends Annotation> annotations) {
     return Arrays.asList(type, annotations);
   }
 
+  /**
+   * Builder class for preparing instances of Moshi.
+   *
+   * Builder instances are thread-safe, meaning multiple threads can safely use a single instance concurrently.
+   */
   public static final class Builder {
-    final List<JsonAdapter.Factory> factories = new ArrayList<>();
+    final List<JsonAdapter.Factory> factories = Collections.synchronizedList(new ArrayList<JsonAdapter.Factory>());
```

Comment:
I can't think of a real reason to support concurrent use of a builder.

Reply:
Me neither. It just seemed like there's no good reason for it not to be thread-safe when the things around it are deliberately designed to be, and it's trivial to make it so. 🤷🏻‍♂️

Reply:
I don't want to commit to thread-safety in the builder because it gives off the wrong impression. Factory order is extremely important so accessing it concurrently means you're opening yourself up to non-deterministic ordering. If you need concurrent access (which will be no one in practice) you can synchronize on the instance you're using.

Reply:
Okay